newlib/libc/time/mktime.c

```diff
@@ -79,19 +79,15 @@ ANSI C requires <<mktime>>.
 #include <time.h>
 #include "local.h"
 
-#define _SEC_IN_MINUTE 60L
-#define _SEC_IN_HOUR 3600L
-#define _SEC_IN_DAY 86400L
-
-static const int DAYS_IN_MONTH[12] =
+static const int8_t DAYS_IN_MONTH[12] =
 {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
 
 #define _DAYS_IN_MONTH(x) ((x == 1) ? days_in_feb : DAYS_IN_MONTH[x])
 
-static const int _DAYS_BEFORE_MONTH[12] =
+static const int16_t _DAYS_BEFORE_MONTH[12] =
 {0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334};
 
-#define _ISLEAP(y) (((y) % 4) == 0 && (((y) % 100) != 0 || (((y)+1900) % 400) == 0))
+#define _ISLEAP(y) isleap((y)+YEAR_BASE)
```

Comment:
Consider changing macros like this to static inline (including 'isleap')?

Reply:
I wanted to do that later, because there is a slightly faster version of isleap() here:
http://howardhinnant.github.io/date_algorithms.html#is_leap

Reply:
That looks the same as the definition of isleap in local.h?

Reply:
The order of the && and || is different. The performance increase is not going to be great.